contents/security/sqlInjection.md

```diff
@@ -92,6 +124,22 @@ SELECT * FROM Users
 
 This will return all the rows from the `Users` table in the database, regardless of username or password entered. If used in user authentication, the user will likely be able to gain access to the system.
 
+### Inject extra queries to mess up the database
+In addition to add extra parameters to compromise the authentication, a malicious user may even add custom queries to do anything to the database.
+
+An SQL query ends with a semicolon ";". In the previous section the malicious user terminates one parameter by single quote "'", and add more parameters behind it. Now, he can also terminate the query by semicolon, and adds another query at the back:
+
+```sql
+Username: foo
+Password: bar’; DROP TABLE Users;
+```
+
+```sql
+SELECT * FROM Users WHERE Username = ‘foo’ AND Password = ‘bar’; 
+DROP TABLE Users;
+```
+
+Then the database will execute these two queries, and deletes all user information. Then other users cannot access this website, and this is the compromise of availability. In addition to `DROP` query, the attacker may also inject `SELECT` and `INSERT` queries which can read data from sensetive database or add data to it. This also compromise confidentiality and integrity respectively.
```

Comment:
'and delete all user'
'and this compromises the availability of data'
'addition to the `DROP`'
'`INSERT` queries, which'
'which can either read'
'sensitive'
'compromises confidentiality and integrity'